### PR TITLE
Method to get and compare running thread id

### PR DESCRIPTION
**Pull request by sebastianbaginski**
@marcusnaslund

### CODE REVIEW

File: source/sdk/threading/Thread.ooc [hunk 1]
```diff
@@ -31,6 +31,24 @@ Thread: abstract class {
...
+	currentThreadId: static func -> Long {
+		version (unix || apple) {
+			return pthread_self() as Long
+		}
+		version (windows) {
+			return GetCurrentThread() as Long
+		}
+		0L
+	}
+	threadIdEquals: static func (threadId1, threadId2: Long) -> Bool {
```
Comment (sebastianbaginski) [inline source/sdk/threading/Thread.ooc]: Any ideas for a better name ?

Comment (marcusnaslund) [inline source/sdk/threading/Thread.ooc]: Maybe just `equals`? There is nothing else to compare with threads.

Comment (sebastianbaginski) [inline source/sdk/threading/Thread.ooc]: OK, why not

**Comment by sebastianbaginski** [comment]
Updated.

File: source/sdk/threading/Thread.ooc [hunk 2]
```diff
@@ -31,6 +31,24 @@ Thread: abstract class {
 		}
 		null
 	}
+	currentThreadId: static func -> Long {
+		version (unix || apple) {
+			return pthread_self() as Long
+		}
+		version (windows) {
+			return GetCurrentThread() as Long
```
Comment (marcusnaslund) [inline source/sdk/threading/Thread.ooc]: I was going to complain but realized the above two methods are like this too. I've made an issue #1261 where I'll try to make files more consistent regarding `return`.

Comment (sebastianbaginski) [inline source/sdk/threading/Thread.ooc]: ok, good idea

File: source/sdk/threading/Thread.ooc [hunk 3]
```diff
@@ -31,6 +31,24 @@ Thread: abstract class {
...
+	}
+	equals: static func (threadId1, threadId2: Long) -> Bool {
+		result: Bool
+		version (unix || apple)
+			result = pthread_equal(threadId1 as PThread, threadId2 as PThread) != 0
+		else
+			result = threadId1 == threadId2
+		result
+	}
+
```
Comment (marcusnaslund) [inline source/sdk/threading/Thread.ooc]: magic complains about two empty lines

Comment (sebastianbaginski) [inline source/sdk/threading/Thread.ooc]: Fixed

**Pull request merged**